src/Form/Form.js

```diff
@@ -97,8 +106,7 @@ class Form {
 
         // checks to determine if we should offer to store credentials and/or fireproof
         const checks = [
-            this.shouldPromptToStoreData,
-            this.hasValues(values),
+            (this.shouldPromptToStoreData && this.hasValues(values)),
```

Comment:
This is a temporary fix for the bug found in the other PR https://github.com/duckduckgo/duckduckgo-autofill/pull/96#discussion_r832146858. It's not pretty, but it doesn't matter because your solution will override it.